src/azul/service/repository_controller.py

```diff
@@ -61,7 +64,11 @@ def download_file(self,
             file_version = file['version']
             drs_path = file['drs_path']
             if file_name is None:
-                file_name = file['name']
+                # To prevent file overwrites due to files with the same name
+                # we insert the crc32c value of the file into the filename.
+                base_name, extension = file_name_partition(file['name'])
+                base_name += '_' + file['crc32c']
```

Comment:
I don't know if you recall @raaidbroad's posts about crc32c, but I think 32 bits is not enough for uniqueness. Can we use filename-safe base64 (without padding) of first 64 bits of sha256 instead? Both are required, right?